doc/design.md

```diff
@@ -0,0 +1,408 @@
+	i := 0
+	/* gofail-label */ myLabel:
+	for i < 5 {
+		s = s + "i"
+		i++
+		for j := 0; j < 5; j++ {
+			s = s + "j"
+			if vExampleLabels, __fpErr := __fp_ExampleLabels.Acquire(); __fpErr == nil { defer __fp_ExampleLabels.Release(); _, __fpTypeOK := vExampleLabels.(struct{}); if !__fpTypeOK { goto __badTypeExampleLabels} 
+				 continue myLabel; __badTypeExampleLabels: __fp_ExampleLabels.BadType(vExampleLabels, "struct{}"); };
+		}
+	}
+	return s
+}
+```
+
+Formatting the code using `gofmt -w .`, then it is below.
+```
+func ExampleLabelsFunc() (s string) {
+	i := 0
+	/* gofail-label */ myLabel:
+	for i < 5 {
+		s = s + "i"
+		i++
+		for j := 0; j < 5; j++ {
+			s = s + "j"
+			if vExampleLabels, __fpErr := __fp_ExampleLabels.Acquire(); __fpErr == nil {
+				defer __fp_ExampleLabels.Release()
+				_, __fpTypeOK := vExampleLabels.(struct{})
+				if !__fpTypeOK {
+					goto __badTypeExampleLabels
+				}
+				continue myLabel
+			__badTypeExampleLabels:
+				__fp_ExampleLabels.BadType(vExampleLabels, "struct{}")
+			}
+		}
+	}
+	return s
+}
+```
+
+**Generated code**:
+```
+var __fp_ExampleLabels *runtime.Failpoint = runtime.NewFailpoint("examples", "ExampleLabels", false)
+```
+
+
+### Example 5: With gofail-go label
+**Original code**:
+```
+func ExampleLabelsGoFunc() (s string) {
+	i := 0
+	// gofail-go: myLabel:
+	for i < 5 {
+		s = s + "i"
+		i++
+		for j := 0; j < 5; j++ {
+			s = s + "j"
+			// gofail-go: var ExampleLabelsGo struct{}
+			// continue myLabel
+		}
+	}
+	return s
+}
+```
+
+The `mylabel` is a gofail-go label, which is used by the customized code `continue myLabel`.
+
+**Translated code**:
+```
+func ExampleLabelsGoFunc() (s string) {
+	i := 0
+	/* gofail-go-label */ myLabel:
+	for i < 5 {
+		s = s + "i"
+		i++
+		for j := 0; j < 5; j++ {
+			s = s + "j"
+			if vExampleLabelsGo, __fpGoErr := __fp_ExampleLabelsGo.Acquire(); __fpGoErr == nil { go __fp_ExampleLabelsGo.Release(); _, __fpTypeOK := vExampleLabelsGo.(struct{}); if !__fpTypeOK { goto __badTypeExampleLabelsGo} 
+				 continue myLabel; __badTypeExampleLabelsGo: __fp_ExampleLabelsGo.BadType(vExampleLabelsGo, "struct{}"); };
+		}
+	}
+	return s
+}
+```
+
+Formatting the code using `gofmt -w .`, then it is below. 
+```
+func ExampleLabelsGoFunc() (s string) {
+	i := 0
+	/* gofail-go-label */ myLabel:
+	for i < 5 {
+		s = s + "i"
+		i++
+		for j := 0; j < 5; j++ {
+			s = s + "j"
+			if vExampleLabelsGo, __fpGoErr := __fp_ExampleLabelsGo.Acquire(); __fpGoErr == nil {
+				go __fp_ExampleLabelsGo.Release()
+				_, __fpTypeOK := vExampleLabelsGo.(struct{})
+				if !__fpTypeOK {
+					goto __badTypeExampleLabelsGo
+				}
+				continue myLabel
+			__badTypeExampleLabelsGo:
+				__fp_ExampleLabelsGo.BadType(vExampleLabelsGo, "struct{}")
+			}
+		}
+	}
+	return s
+}
+```
+
+There are two differences between the translated "gofail" and "gofail-go" code,
+1. "gofail" uses `defer` to call `Release()`, while "gofail-go" uses `go` to call the method;
```

Comment:
Another nit, use of `;` vs `.` in the second point as sentence ending. Since they are two different sentences, replace `;` with dot. ?

Reply:
thx & fixed